CONTRIBUTING.md

```diff
@@ -51,9 +51,42 @@ Once we've discussed your changes and you've got your code ready, make sure that
 
 ## Contribution Guidelines for Documentation
 
+* pre-commit
+
+    A framework for managing and maintaining multi-language pre-commit hooks.
+    Pre-commit can be [installed](https://pre-commit.com/#installation) with `pip`, `curl`, `brew` or `conda`.
+    You need to first install pre-commit and then install the pre-commit hooks with `pre-commit install`.
+    Now pre-commit will run automatically on git commit!
```

Comment:
Suggest removing this line since it's not related to contributing guide

Reply:
Having as much information as possible on `pre-commit` is necessary.   
So far both you and @spacewander don't seem to understand much about `pre-commit`. 
Perhaps we need a new Markdown file for these docs ?